### PR TITLE
fix some Ad-Shield recovery

### PR DESCRIPTION
**Pull request by JobcenterTycoon**
https://listauthorschat.slack.com/archives/C05G8V3NERH/p1716909603510399?thread_ts=1716891677.627739&cid=C05G8V3NERH

### CODE REVIEW

File: BaseFilter/sections/specific.txt [hunk 1]
```diff
@@ -20,6 +20,16 @@ triblive.com###storyContent > div[data-type="float"][data-stn-player]
 ! !SECTION: Temporary
 !##################
 !
+! Ad-Shield ad reinsertion
+fourfourtwo.co.kr,interfootball.co.kr,a-ha.io,cboard.net,jjang0u.com,joongdo.co.kr,viva100.com,thephoblographer.com,gamingdeputy.com,thesaurus.net,tweaksforgeeks.com,alle-tests.nl,maketecheasier.com,automobile-catalog.com,allthekingz.com,motorbikecatalog.com#%#//scriptlet('abort-current-inline-script', 'atob', 'eval(decodeURIComponent(escape(window.atob("KCgpPT57bGV0IGU')
+fourfourtwo.co.kr,interfootball.co.kr,a-ha.io,cboard.net,jjang0u.com,joongdo.co.kr,viva100.com,thephoblographer.com,gamingdeputy.com,thesaurus.net,tweaksforgeeks.com,alle-tests.nl,maketecheasier.com,automobile-catalog.com,allthekingz.com,motorbikecatalog.com#%#//scriptlet('prevent-eval-if', '/07c225f3\.online|content-loader\.com|css-load\.com|html-load\.com/')
+fourfourtwo.co.kr,interfootball.co.kr,a-ha.io,cboard.net,jjang0u.com,joongdo.co.kr,viva100.com,thephoblographer.com,gamingdeputy.com,thesaurus.net,tweaksforgeeks.com,alle-tests.nl,maketecheasier.com,automobile-catalog.com,allthekingz.com,motorbikecatalog.com$$script[tag-content="KCgpPT57bGV0IGU"]
```
Comment (Alex-302) [inline BaseFilter/sections/specific.txt:26]: Btw have updated `$$` syntax https://adguard.com/kb/general/ad-filtering/create-own-filters/#pseudo-classes

```suggestion
fourfourtwo.co.kr,interfootball.co.kr,a-ha.io,cboard.net,jjang0u.com,joongdo.co.kr,viva100.com,thephoblographer.com,gamingdeputy.com,thesaurus.net,tweaksforgeeks.com,alle-tests.nl,maketecheasier.com,automobile-catalog.com,allthekingz.com,motorbikecatalog.com$$script:contains(KCgpPT57bGV0IGU)
```

File: BaseFilter/sections/specific.txt [hunk 2]
```diff
@@ -20,6 +20,16 @@ triblive.com###storyContent > div[data-type="float"][data-stn-player]
...
+worldhistory.org,etnews.com,iusm.co.kr,lamire.jp,picrew.me,text-compare.com,etnewskorea.com,gadgets360.com,mydaily.co.kr,ndtvprofit.com,badmouth1.com,logicieleducatif.fr,taxguru.in$$script[onload="error-report.com"]
+allthetests.com,javatpoint.com,globalrph.com#%#//scriptlet('remove-node-text', 'script', 'error-report.com')
+allthetests.com,javatpoint.com,globalrph.com$$script[tag-content="error-report.com"]
```
Comment (Alex-302) [inline BaseFilter/sections/specific.txt:29]:
```suggestion
allthetests.com,javatpoint.com,globalrph.com$$script:contains(error-report.com)
```

File: BaseFilter/sections/specific.txt [hunk 3]
```diff
@@ -20,6 +20,16 @@ triblive.com###storyContent > div[data-type="float"][data-stn-player]
...
+worldhistory.org#%#//scriptlet('remove-node-text', 'script', 'adShield')
+worldhistory.org$$script[tag-content="adShield"]
```
Comment (Alex-302) [inline BaseFilter/sections/specific.txt:31]:
```suggestion
worldhistory.org$$script:contains(adShield)
```

**Review by Alex-302 — CHANGES_REQUESTED** [review]

**Comment by Alex-302** [comment]
@piquark6046 Please also take a look.

**Review by piquark6046 — CHANGES_REQUESTED** [review]
please nest them with `!#if (!adguard_ext_safari && !adguard_app_ios && !adguard_ext_android_cb)`.

**Review by piquark6046 — COMMENTED** [review]
I believe that Ad-Shield will update their logic to interfere the filtering when a AdGuard filter handles it.
https://web.archive.org/web/20221005080821/https://www.ad-shield.team/blog/how-adshield-fight
https://web.archive.org/web/20220930005843/https://docs.ad-shield.io/update-notes

**Comment by Alex-302** [comment]
@piquark6046 @JobcenterTycoon 

> I believe that Ad-Shield will update their logic to interfere the filtering when a AdGuard filter handles it.

But uBO has similar rules?

**Comment by piquark6046** [comment]
https://github.com/AdguardTeam/AdguardFilters/pull/180287#issuecomment-2137232852
> But uBO has similar rules?

It seems that they focus more on AdGuard than uBO.

**Comment by Alex-302** [comment]
> It seems that they focus more on AdGuard than uBO.

Hmmm. At least we can try)

**Comment by piquark6046** [comment]
> Hmmm. At least we can try)

Actually, I afraid of a widespread breakage that occurs after Ad-Shield dev interferes this changes.

**Comment by JobcenterTycoon** [comment]
They didn't circumvented the spyware filter even if the site switched to ad reinsertion long time ago already.

**Comment by piquark6046** [comment]
Then, do we think that they focus more on ABP than AdGuard and uBO because their offline office exists in USA where ABP is most used?

**Comment by Yuki2718** [comment]
> adguard_ext_android_cb does not support these rules. No need to exclude.

Maybe replace with `ext_ublock`, as redundant and can cause trouble in future.

**Comment by JobcenterTycoon** [comment]
I did

File: BaseFilter/sections/specific.txt [hunk 4]
```diff
@@ -20,6 +20,18 @@ triblive.com###storyContent > div[data-type="float"][data-stn-player]
 ! !SECTION: Temporary
 !##################
 !
+! Ad-Shield ad reinsertion
+!#if (!adguard_ext_safari && !adguard_app_ios && !adguard_ext_android_cb && !ext_ublock)
```
Comment (Alex-302) [inline BaseFilter/sections/specific.txt:24]:
```suggestion
!#if (!adguard_ext_safari && !adguard_app_ios && !ext_ublock)
```

`AdGuard Content Blocker` (not AG for Android) can apply only basic rules. `$$` or `#%#` and others automatically removed from filter version for AdGuard Content Blocker.

Btw you can use this button if suggestion is ok

image

and/or press `Resolve conversation`.

**Review by Alex-302 — CHANGES_REQUESTED** [review]

File: BaseFilter/sections/specific.txt (hunk 4 above)
Comment (JobcenterTycoon) [reply BaseFilter/sections/specific.txt:24]: Hmm nothing happened after i pressed the button. Changed manually now.

Comment (Alex-302) [reply BaseFilter/sections/specific.txt:24]: Hm. Weird.
Thanks.

**Review by Alex-302 — APPROVED** [review]

**Comment by Alex-302** [comment]
@piquark6046 Merged.

File: BaseFilter/sections/specific.txt (hunk 4 above)
Comment (Yuki2718) [reply BaseFilter/sections/specific.txt:24]: > `AdGuard Content Blocker` (not AG for Android) can apply only basic rules. `$$` or `#%#` and others automatically removed from filter version for AdGuard Content Blocker.

Might be safer to include unless we are sure we never add classic rules, as applying only classic rules without advanced ones can harm.

**Pull request merged**